cmd/litcli/ln.go

```diff
@@ -237,74 +236,80 @@ type resultStreamWrapper struct {
+	printQuote := func(quote *rfqrpc.PeerAcceptedSellQuote) error {
 		rpcRate := quote.BidAssetRate
 		rate, err := rpcutils.UnmarshalRfqFixedPoint(rpcRate)
 		if err != nil {
-			return nil, fmt.Errorf("unable to unmarshal fixed "+
-				"point: %w", err)
+			return fmt.Errorf("unable to unmarshal fixed point: %w",
+				err)
 		}
 
 		amountMsat := lnwire.MilliSatoshi(w.amountMsat)
 		milliSatsFP := rfqmath.MilliSatoshiToUnits(amountMsat, *rate)
 		numUnits := milliSatsFP.ScaleTo(0).ToUint64()
 
-		// If the calculated number of units is 0 then the asset rate
-		// was not sufficient to represent the value of this payment.
+		// The purpose of this function is just to print, so let's avoid
+		// dividing by zero or reporting an invalid msat/unit rate.
 		if numUnits == 0 {
-			// We will calculate the minimum amount that can be
-			// effectively sent with this asset by calculating the
-			// value of a single asset unit, based on the provided
-			// asset rate.
-
-			// We create the single unit.
-			unit := rfqmath.FixedPointFromUint64[rfqmath.BigInt](
-				1, 0,
-			)
-
-			// We derive the minimum amount.
-			minAmt := rfqmath.UnitsToMilliSatoshi(unit, *rate)
-
-			// We return the error to the user.
-			return nil, fmt.Errorf("smallest payment with asset "+
-				"rate %v is %v, cannot send %v",
-				rate.ToUint64(), minAmt, amountMsat)
+			return nil
 		}
 
 		msatPerUnit := uint64(w.amountMsat) / numUnits
 
 		fmt.Printf("Got quote for %v asset units at %v msat/unit from "+
-			"peer %s with SCID %d\n", numUnits, msatPerUnit,
+			" peer %s with SCID %d\n", numUnits, msatPerUnit,
 			quote.Peer, quote.Scid)
 
-		resp, err = w.stream.Recv()
+		return nil
+	}
+
+	// A boolean to indicate whether the first quote was printed via the
+	// legacy single-rfq response field.
+	legacyFirstPrint := false
+
+	for {
+		resp, err := w.stream.Recv()
 		if err != nil {
 			return nil, err
 		}
 
-		if resp == nil || resp.Result == nil ||
-			resp.GetPaymentResult() == nil {
+		res := resp.Result
 
-			return nil, errors.New("unexpected nil result")
-		}
+		switch r := res.(type) {
+		case *tchrpc.SendPaymentResponse_AcceptedSellOrder:
+			err := printQuote(r.AcceptedSellOrder)
+			if err != nil {
+				return nil, err
+			}
 
-		return resp.GetPaymentResult(), nil
+			legacyFirstPrint = true
 
-	case *tchrpc.SendPaymentResponse_PaymentResult:
-		return r.PaymentResult, nil
+		case *tchrpc.SendPaymentResponse_AcceptedSellOrders:
+			quotes := r.AcceptedSellOrders.AcceptedSellOrders
 
-	default:
-		return nil, fmt.Errorf("unexpected response type: %T", r)
+			for _, quote := range quotes {
+				// If the first item was returned via the legacy
+				// field then skip printing it again here. This
+				// skip only applies to the first element.
+				if legacyFirstPrint {
+					legacyFirstPrint = false
+					continue
+				}
+
+				err := printQuote(quote)
+				if err != nil {
+					return nil, err
+				}
+			}
```

Comment:
Just checking that this is correct behaviour:
There is no `return` statement for the `*tchrpc.SendPaymentResponse_AcceptedSellOrder` & `*tchrpc.SendPaymentResponse_AcceptedSellOrders` cases here unless there's no error, meaning that they won't break the for loop without any errors being thrown somewhere in the next loop iteration. Is that correct behaviour?